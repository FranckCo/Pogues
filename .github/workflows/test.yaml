name: Pogues Test CI

on:
  push:
    branches:
      - '**'

jobs:
  test-build:
    name: Test & build
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: 20
      - run: yarn
      - run: yarn test:coverage
      - run: yarn build
      - name: Upload coverage artifact
        uses: actions/upload-artifact@v4
        with:
          name: coverage
          path: coverage/
      - name: Upload build artifact
        uses: actions/upload-artifact@v4
        with:
          name: build
          path: dist/

  sonarcloud:
    name: SonarCloud Analysis
    runs-on: ubuntu-latest
    needs: test-build # Ensure SonarCloud job runs after the test-build job
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0

      - name: Download coverage artifact
        uses: actions/download-artifact@v4
        with:
          name: coverage
          path: coverage

      - name: SonarCloud Scan
        uses: sonarsource/sonarcloud-github-action@v1.13
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        with:
          args: >
            -Dsonar.pullrequest.key=${{ github.event.pull_request.number }}
            -Dsonar.pullrequest.branch=${{ github.head_ref }}
            -Dsonar.pullrequest.base=${{ github.base_ref }}
