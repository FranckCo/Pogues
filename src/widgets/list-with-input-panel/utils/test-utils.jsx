export function getFakeFields(items, inputName) {
  return {
    name: 'FAKE_NAME.FAKE_NAME',
    length: items.length,
    items: items,
    fields: items.map((it, index) => {
      return {
        index,
        name: `${inputName}[${index}]`,
        listFields: {
          get: indexInner => items[indexInner],
        },
      };
    }),
    map(func) {
      return items.map((it, index) => {
        const field = this.fields[index];
        return func.call(this, field.name, field.index, field.listFields);
      });
    },
  };
}
