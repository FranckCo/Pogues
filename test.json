{"id":"cl_sexe","name":"cl_sexe","label":"Sexe","codes":[{"value":0,"label":"Homme"},{"value":1,"label":"Femme"}]}