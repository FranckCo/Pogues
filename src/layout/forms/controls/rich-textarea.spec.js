import React from 'react';
import { decoreFormField } from 'utils/test/test-utils';
import { Field } from 'redux-form';
import TextArea from './rich-textarea';

describe('RichTextArea', () => {
  test('should have this default template', () => {
    const tree = decoreFormField(<Field component={TextArea} name="checkbox" label="Label Name" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have a help block', () => {
    const tree = decoreFormField(<Field help component={TextArea} name="checkbox" label="Label Name" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
