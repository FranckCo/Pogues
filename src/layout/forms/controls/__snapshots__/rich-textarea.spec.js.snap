// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextArea should have a help block 1`] = `
<div
  className="ctrl-input"
>
  <label
    htmlFor="select-checkbox"
  >
    Label Name
    
     
    <span
      className="help-block"
    >
      <span
        aria-hidden="true"
        className="glyphicon glyphicon-question-sign"
      />
       
      Help
       
    </span>
  </label>
  <div>
    <textarea
      id="select-checkbox"
      name="checkbox"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      value=""
    />
  </div>
</div>
`;

exports[`RichTextArea should have this default template 1`] = `
<div
  className="ctrl-input"
>
  <label
    htmlFor="select-checkbox"
  >
    Label Name
    
     
  </label>
  <div>
    <textarea
      id="select-checkbox"
      name="checkbox"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      value=""
    />
  </div>
</div>
`;
